client/src/lib/dependency-injection/withInjector.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+function withInjector(Component, dependencies, mapDependenciesToProps) {
+  const injectorContext = {
+    injector: React.PropTypes.shape({
+      get: React.PropTypes.func,
+    }),
+  };
+
+  class Injectable extends React.Component {
+    render() {
+      let props = {};
+      if (dependencies) {
+        if (!Array.isArray(dependencies)) {
+          throw new Error(`
+            withInjector() passed an argument for dependencies that is ${typeof dependencies}. 
+            Must be an array of named dependencies.
+          `);
+          return;
+        }
+        const resolved = dependencies.map(this.context.injector.get);
```

Comment:
So the constructed container could declare dependencies as propTypes, and this HoC resolves them automatically. Does that mean most container components and some presentational components would be wrapped this way in their default export? I assume that doesn't interfere with the ability to further wrap these exported components with `customise()`?